code-scanning/trivy.yml

```diff
@@ -14,8 +14,13 @@ on:
   schedule:
     - cron: $cron-weekly
 
+permissions: read-all
+
 jobs:
   build:
+    permissions:
+      contents: read # for actions/checkout to fetch code
```

Comment:
Hi, I'm a maintainer of Trivy. I'd like to know why this line is needed. Looks like `read-all` is granted in the top-level permissions.

Reply:
Good question. If you define permissions in the job, those you do not specify default to `none`, not to the ones defined at the top level. The top level permissions are therefore a fail-safe backup if a developer ever adds a new job and forget to set permissions explicitely.